### ISSUE TITLE
Replicated Softmax does not work well on documents with different length

### ISSUE THREAD

**Issue by BayesianHuman** [AmazaspShumik/Bayesian-Regression-Methods#22]

**Comment by AmazaspShumik** [comment]
@BayesianHuman 

Hi, I was trying to implement Replicated Softmax, but neither scaled version (where hidden units are scaled by the length of a document) nor standard version works well. I tried to repeat the same experiment as Hinton and Salakhutdinov, but was not able to achieve the same results
(most likely because of difference in text preprocessing) and in many cases, LDA features were superior to those found by RSM. Since I need to refactor / improve other algorithms in this package
I decided to remove ReplicatedSoftmax.

There are several other implementations available ( I still was not able to get good results with them, but maybe they will be useful):
https://github.com/arongdari/RSM/blob/master/oRSM.py)

**Comment by BayesianHuman** [comment]
Thanks for the suggestion, I will look into that implementation!

Can you explain in detail what text preprocessing did you do? 
For instance, did you remove stop words? Did you do stemming? How did you compare LDA vs. RSM, was it by retrieval accuracy?

**Comment by AmazaspShumik** [comment]
1) I compared using retrieval accuracy ( similarity measure = cosine similarity as in RSM article)
2) I used CountVectorizer from scikit-learn, which has some internal set of stop words.
3) I think that CountVectorizer does not do any stemming, so it seems that I did not do that

**Comment by AmazaspShumik** [comment]
@BayesianHuman 

I am closing this issue if you are not against.

**Comment by BayesianHuman** [comment]
Thanks for the help !
